hazelcast-code-generator/src/main/resources/codec-template-cpp.ftl

```diff
@@ -186,7 +186,7 @@ namespace hazelcast {
                     dataSize += ClientMessage::calculateDataSize<${util.getCppType(itemVariableType)} >(${varName});
                 <#break >
             <#case "ARRAY">
-                <#local itemVariableType= util.getGenericType(type)>
+                <#local itemVariableType= util.getArrayType(type)>
```

Comment:
Do we have any kind of tests for this? Where do we see if something like this is wrong?